pipeline.go

```diff
@@ -345,59 +345,96 @@ func (p *PipelineDefinition) Pipelines() (*Pipelines, error) {
 	return p.pipelines, nil
 }
 
-func runParallelBuildImage(runner Runner, step Step, pull bool, durations *sync.Map, wg *sync.WaitGroup, start chan struct{}, abort chan struct{}) {
-	defer wg.Done()
-	<-start
+type runConfig struct {
+	usePreconditions bool
```

Comment:
Can't this information be encoded as just a `nil` `pre` func?

Reply:
Yes it can and will be, I just didn't think about that.